packages/dm-core-plugins/src/file/FilePlugin.tsx

```diff
@@ -0,0 +1,202 @@
+import * as React from 'react'
+
+import {
+  ErrorResponse,
+  IUIPlugin,
+  Loading,
+  TGenericObject,
+  useDMSS,
+} from '@development-framework/dm-core'
+import { useDocument } from '@development-framework/dm-core'
+import { AxiosError, AxiosRequestConfig } from 'axios'
+import { Button, Progress } from '@equinor/eds-core-react'
+import { ChangeEvent, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import { useRegistryContext } from '../form/RegistryContext'
+
+export const ErrorGroup = styled.div`
+  display: flex;
+  flex-direction: column;
+  border: 1px solid rgba(213, 18, 18, 0.71);
+  border-radius: 5px;
+  padding: 20px 20px;
+  background-color: #f6dfdf;
+`
+
+const UploadFileButton = (props: any) => {
```

Comment:
```suggestion
const UploadFileButton = (props: {onUpload: (file ?, reference: TLinkReference) => void}) => {
```

not sure what type of file is. 


Without adding type to props this plugin is very difficult to use

Reply:
fixed